File: examples/angular2/dart/lib/fela_renderer_service.dart
import 'package:angular2/core.dart';
import 'dart:html';
import 'package:fela_example_angular_2_dart/fela_js_bridge.dart' as Fela;

@Injectable()
class FelaRendererService {
  static Fela.Renderer renderer = null;

  constructor() {
    if (FelaRendererService.renderer == null) {
      FelaRendererService.renderer = Fela.createRenderer();
      Fela.render(renderer, querySelector('#felaStyles'));
    }
  }

  String renderRule(Function rule, Map<String, dynamic> props) {
    return FelaRendererService.renderer.renderRule(rule, props);
  }

  clear() {
    FelaRendererService.renderer.clear();
  }

}

File: examples/angular2/typescript/app/felaRenderer.service.ts
import { Injectable } from '@angular/core';
import { createRenderer, render, Renderer } from 'fela';

@Injectable()
export class FelaRendererService {
  private static renderer:Renderer = null;

  constructor() {
    if (FelaRendererService.renderer === null) {
      FelaRendererService.renderer = createRenderer();
      render(FelaRendererService.renderer, document.getElementById('felaStyles'));
    }
  }

  public renderRule(rule: Function, props?: Object): string {
    return FelaRendererService.renderer.renderRule(rule, props);
  }

}
